src/core/read/py_fread.cc

```diff
@@ -37,20 +37,201 @@ static const char* doc_fread =
 R"(fread(anysource=None, *, file=None, text=None, cmd=None, url=None,
          columns=None, sep=None, dec=".", max_nrows=None, header=None,
          na_strings=None, verbose=False, fill=False, encoding=None,
-         skip_to_string=None, skip_to_line=None, skip_blank_lines=False,
-         strip_whitespace=True, quotechar='"', save_to=None,
-         tempdir=None, nthreads=None, logger=None, multiple_sources="warn",
+         skip_to_string=None, skip_to_line=0, skip_blank_lines=False,
+         strip_whitespace=True, quotechar='"', tempdir=None,
+         nthreads=None, logger=None, multiple_sources="warn",
          memory_limit=None)
 --
 
+This function is capable of reading data from a variety of input formats,
+producing a :class:`Frame` as the result. The recognized formats are:
+CSV, Jay, XLSX, and plain text. In addition, the data may be inside an
+archive such as .tar, .gz, .zip, .gz2.
+
+
+Parameters
+----------
+anysource: str | bytes | file | Pathlike | List
+    The first (unnamed) argument to fread is the *input source*.
+    Multiple types of sources are supported, and they can be named
+    explicitly: `file`, `text`, `cmd`, and `url`. When the source is
+    not named, fread will attempt to guess its type. The most common
+    type is `file`, but sometimes the argument is resolved as `text`
+    (if the string contains newlines) or `url` (if the string starts
+    with `https://` or similar).
+
+    Only one argument out of `anysource`, `file`, `text`, `cmd` or
+    `url` can be specified at once.
+
+file: str | file | Pathlike
+    A file source can be either the name of the file on disk, or a
+    python "file-like" object -- i.e. any object having method
+    ``.read()``.
```

Comment:
We might want to add that the "file-like" objects won't work on Windows.

Reply:
This should work on Windows too. See `tests/fread/test_fread_api.py::test_fread_from_fileobj` and `tests/fread/test_fread_api.py::test_fread_from_anysource_filelike`.

Reply:
`tests/fread/test_fread_api.py::test_fread_from_fileobj` was crashing on Windows without setting any errors and that was because of the wrong file descriptor. That's why we had this PR https://github.com/h2oai/datatable/pull/2412